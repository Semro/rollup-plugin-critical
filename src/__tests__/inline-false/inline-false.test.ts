import fs from 'fs';
import path from 'path';
import PluginCritical from '../../index';
import {Plugin} from 'rollup';
import {expect, test} from 'vitest'

const testRoot = path.join(__dirname, '/');
const testOutputPath = path.join(testRoot, '__output__/test_critical.min.css');

const pluginConfig: CriticalPluginConfig = {
    criticalBase: testRoot,
    criticalUrl: testRoot,
    criticalPages: [
        {
            uri: 'index.html',
            template: '__output__/test',
        }
    ],
    criticalConfig: {
        inline: false,
    },
};

test('`inline: false` Critical CSS generation', async () => {
    // Instantiate the Rollup plugin
    const plugin: Plugin = PluginCritical(pluginConfig);
    // Call the plugin to generate critical css
    if (plugin && typeof plugin.writeBundle === 'function') {
        // @ts-ignore
        await plugin.writeBundle({
            dir: testRoot,
        }, {
            chunk: {
                type: 'asset',
                fileName: 'style.css',
            }
        });
        // Compare the output with the snapshot
        expect(fs.readFileSync(testOutputPath).toString())
          .toMatchSnapshot();
    }
});
