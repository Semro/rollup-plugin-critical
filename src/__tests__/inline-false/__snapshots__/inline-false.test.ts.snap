// Vitest Snapshot v1

exports[`\`inline: false\` Critical CSS generation 1`] = `".heading{color:red}"`;
